circleci.sh

```diff
@@ -0,0 +1,120 @@
+            break
+        elif [ $i -ge 4 ]; then
+            sleep $((1 << $i))
+        else
+            echo 'Failed to download install script' 1>&2
+            exit 1
+        fi
+    done
+
+    source "$(CURL_USER_AGENT=\"$CURL_USER_AGENT\" bash install.sh dmd-$HOST_DMD_VER --activate)"
+    $DC --version
+    env
+}
+
+# clone dmd and druntime
+clone() {
+    local url="$1"
+    local path="$2"
+    local branch="$3"
+    for i in {0..4}; do
+        if git clone --branch "$branch" "$url" "$path" "${@:4}"; then
+            break
+        elif [ $i -lt 4 ]; then
+            sleep $((1 << $i))
+        else
+            echo "Failed to clone: ${url}"
+            exit 1
+        fi
+    done
+}
+
+install_make()
+{
+    mkdir -p make
+    cd make
+    curl -L http://ftp.gnu.org/gnu/make/make-4.1.tar.gz  | tar -zxf - --strip-components=1 && ./configure && make && ./make -v
+    export PATH="$(pwd):${PATH}"
+    cd ..
+}
+
+setup_repos()
+{
+    # Set a default in case we run into rate limit restrictions
+    local base_branch=""
+    if [ -n "${CIRCLE_PR_NUMBER:-}" ]; then
+        base_branch=$((curl -fsSL https://api.github.com/repos/dlang/phobos/pulls/$CIRCLE_PR_NUMBER || echo) | jq -r '.base.ref')
+    else
+        base_branch=$CIRCLE_BRANCH
+    fi
+    base_branch=${base_branch:-"master"}
+
+    # Merge upstream branch with changes, s.t. we check with the latest changes
+    if [ -n "${CIRCLE_PR_NUMBER:-}" ]; then
+        local current_branch=$(git rev-parse --abbrev-ref HEAD)
+        git config user.name dummyuser
+        git config user.email dummyuser@dummyserver.com
+        git remote add upstream https://github.com/dlang/phobos.git
+        git fetch upstream
+        git checkout -f upstream/$base_branch
+        git merge -m "Automatic merge" $current_branch
+    fi
+
+    for proj in dmd druntime phobos ; do
+        if [ $base_branch != master ] && [ $base_branch != stable ] &&
+            ! git ls-remote --exit-code --heads https://github.com/dlang/$proj.git $base_branch > /dev/null; then
+            # use master as fallback for other repos to test feature branches
+            clone https://github.com/dlang/$proj.git ../$proj master --depth 1
+        else
+            clone https://github.com/dlang/$proj.git ../$proj $base_branch --depth 1
+        fi
+    done
+
+    # Install the bootstrap compiler
+    CURL_USER_AGENT=\"$CURL_USER_AGENT\" bash ~/dlang/install.sh dmd-$HOST_DMD_VER --activate
+}
+
+run_make()
+{
+    # Load environment for bootstrap compiler
+    source "$(CURL_USER_AGENT=\"$CURL_USER_AGENT\" bash ~/dlang/install.sh dmd-$HOST_DMD_VER --activate)"
+    export PATH="$(pwd)/make:$PATH"
+    make -v
+
+    # TODO: d-release.tag target is currently broken
+    #make -f posix.mak RELEASE=1 release -j5 html dmd-release druntime-release phobos-release d-release.tag
```

Comment:
So this already shows that setting up a CI is _very_ important. I will try to fix this in a follow-up PR once this is merged.